alcs-api/src/application/application-time-tracking.service.ts

```diff
@@ -0,0 +1,80 @@
+import { Injectable } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Repository } from 'typeorm';
+import { BusinessDayService } from '../providers/business-days/business-day.service';
+import { ApplicationPaused } from './application-paused.entity';
+import { Application } from './application.entity';
+
+export type ApplicationTimeData = {
+  activeDays: number;
+  pausedDays: number;
+};
+
+@Injectable()
+export class ApplicationTimeTrackingService {
+  constructor(
+    @InjectRepository(ApplicationPaused)
+    private applicationPausedRepository: Repository<ApplicationPaused>,
+    private businessDayService: BusinessDayService,
+  ) {}
+
+  async fetchApplicationActiveTimes(applications: Application[]) {
+    const appUuids = applications.map((app) => app.uuid);
+    const pausedTimes = await this.getApplicationPausedTimes(appUuids);
+
+    const resultMap = new Map<string, ApplicationTimeData>();
+
+    applications.forEach((app) => {
+      const totalBusinessDays = this.businessDayService.calculateDays(
+        new Date(app.createdAt),
+        new Date(),
+      );
+      const pausedDays = pausedTimes.get(app.uuid);
+
+      resultMap.set(app.uuid, {
+        pausedDays: pausedDays || 0,
+        activeDays: totalBusinessDays - pausedDays,
+      });
+    });
+
+    return resultMap;
+  }
+
+  private async getApplicationPausedTimes(applicationUuids: string[]) {
+    const pausedTimes = (await this.applicationPausedRepository.query(
+      `
+        SELECT
```

Comment:
Is it here as a temporary solution? If yes ignore, otherwise
what if you create a function on db level and then call it here? This way we can keep this things separate and later test them.

Reply:
Done! Moved to a function and added a migration for it